tests/framework/helpers/FileHelperTest.php

```diff
@@ -138,6 +138,85 @@ public function testCopyDirectory()
         }
     }
 
+    public function testCopyDirectoryRecursive()
+    {
+        $srcDirName = 'test_src_dir_rec';
+        $structure = [
+            'directory1' => [
+                'file1.txt' => 'file 1 content',
+                'file2.txt' => 'file 2 content',
+            ],
+            'directory2' => [
+                'file3.txt' => 'file 3 content',
+                'file4.txt' => 'file 4 content',
+            ],
+            'file5.txt' => 'file 5 content',
+        ];
+        $this->createFileStructure([
+            $srcDirName => $structure
+        ]);
+
+        $basePath = $this->testFilePath;
+        $srcDirName = $basePath . DIRECTORY_SEPARATOR . $srcDirName;
+        $dstDirName = $basePath . DIRECTORY_SEPARATOR . 'test_dst_dir';
+
+        FileHelper::copyDirectory($srcDirName, $dstDirName);
+
+        $this->assertFileExists($dstDirName, 'Destination directory does not exist!');
+
+        $checker = function ($structure, $dstDirName) use (&$checker) {
```

Comment:
Why it can't be done without putting it in a closure? It's not a problem or principle question, I'm just asking to know, what is the idea of this approach.